ls1intum/Artemis
# `Notifications`: Fix small issues related to email notifications

### Checklist
#### General
#### Server (The majority of changes are due to new tests. Logic is almost untouched)
#### Client (Only minimal renamings or type changes)

### Motivation and Context
I collected minimal change requests for the new email support for notifications (mostly concerning the code) and implemented them here. I also finally added/expanded test classes for the most important notification classes on the server-side (which had no tests so far (legacy-issue))

### Description
Most changes are very minimal typos, type changes, or renamings. The biggest part of this PR (LoC) are the new tests.
For the tests, I tried to cover as many functionalities as possible and do so in the fashion of the already merged tests for GroupNotificationFactory and GroupNotificationService (also written by me).

### Steps for Testing
Mostly look at the code.
(Feel free to execute the test suites locally if you want to)
Try to play around with the notification settings and emails to make sure everything still works (which is/should be the case because the changes to the logic are minimal)

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
GroupNotificationService : ~20% -> 100%    (time : < 1 s)
SingleUserNotificationService : 0 ->  ~100% (time : ~500 ms)
SingleUserNotificationFactory : 0 -> ~100%  (time : ~ 400 ms)
